### PR TITLE
Wadm 0.4 Release Workflow Update

### PR DESCRIPTION
**Pull request by stephenandary**
## Feature or Problem
Updating release workflow for wadm 0.4.

## Related Issues
<https://github.com/wasmCloud/wadm/issues/54>

## Testing

Built on platform(s)
- [x] x86_64-linux
- [x] aarch64-linux
- [x] x86_64-darwin
- [x] aarch64-darwin
- [x] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,157 +1,232 @@
...
-
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
+      - name: setup for cross-compile builds
+        if: matrix.config.arch == 'aarch64' && matrix.config.os == 'ubuntu-20.04'
+        run: |
+          sudo apt-get update
+          sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu
+          rustup toolchain install stable-aarch64-unknown-linux-gnu
```
Comment (stephenandary) [inline .github/workflows/release.yml:96]: I have tried to condense this step with the following "Install Rust Stable Toolchain Step", however, it does not work. Not really sure why, and don't want to spend much more time on this. I have tried changing step order as well. For now this works, but it seems redundant to have this step. May be nice to condense in future.

**Comment by stephenandary** [comment]
I have removed steps related to actions-rs in the workflow, as actions-rs is no longer maintained. For toolchain installation, I have switched to Dtolnay/Toolchain and for actions-rs/cargo I am just now running cargo commands with `run:`

**Comment by stephenandary** [comment]
Nothing has been tested yet, but the PR is at a good point now to come out of Draft to discuss.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,157 +1,236 @@
 name: Wadm Release
-
 on:
   push:
-    tags: ["v*"] # Push events to matching v*, i.e. v1.0, v20.15.10
+    tags: 
+      - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
+      - "canary"
```
Comment (thomastaylor312) [inline .github/workflows/release.yml:6]: By default a push to main should be `canary` and just overwrite the current version. That is basically what you have below, but we aren't planning on pushing just a plain canary tag

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -1,157 +1,236 @@
...
+
+      - name: set the release version (main)
+        if: github.ref == 'refs/heads/main'
+        shell: bash
+        run: echo "RELEASE_VERSION=canary" >> $GITHUB_ENV
 
-      - name: Create Mix Release
-        working-directory: ${{env.working-directory}}
+      # To manually test workflow
+      - name: set the release version (manually trigger test with canary tag push)
```
Comment (thomastaylor312) [inline .github/workflows/release.yml:69]: If you want to keep this around, you can just have a workflow trigger

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+RUN addgroup -g $USER_GID $USERNAME \
+    && adduser -D -u $USER_UID -G $USERNAME $USERNAME
+
+# Copy application binary from builder image
+COPY --from=builder --chown=$USERNAME /usr/src/wadm/target/x86_64-unknown-linux-musl/release/wadm /usr/local/bin/wadm
+
+USER $USERNAME
+# Run the application
+CMD ["/usr/local/bin/wadm"]
```
Comment (thomastaylor312) [inline Dockerfile:27]: Can you change this to be `ENTRYPOINT` so that people can pass args to the container?

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]

**Review by thomastaylor312 — APPROVED** [review]

**Comment by thomastaylor312** [comment]
Great work here @stephenandary!

**Comment by thomastaylor312** [comment]
Also, please note that this will not run until we merge `wadm_0.4` into `main` and that is fine

**Pull request merged**